discovery.go

```diff
@@ -12,6 +15,38 @@ import (
 	handling "github.com/coreos/discovery.etcd.io/http"
 )
 
+func fail(err string) {
+	log.Print(err)
+	pflag.PrintDefaults()
+	os.Exit(2) // default go flag error code
+}
+
+func getValidUrlOrExit(givenUrl string) string {
```

Comment:
the idiomatic go name for this sort of pattern would start with `must`-- `func mustHostOnlyURL(givenUrl string) string`?